packages/flutter/lib/src/foundation/memory_allocations.dart

```diff
@@ -18,6 +18,39 @@ const bool _kMemoryAllocations = bool.fromEnvironment('flutter.memory_allocation
 /// `--dart-define=flutter.memory_allocations=true`.
 const bool kFlutterMemoryAllocationsEnabled = _kMemoryAllocations || kDebugMode;
 
+/// Create an object with exemption from release.
+///
+/// All objects created by [builder] will be exempt from release.
+///
+/// Exempt from release for an object means the object is
+/// not expected to be disposed and garbage collected
+/// during application lifecycle. However, if the object is
+/// garbage collected without being disposed, it is considered as
+/// violation of disposal contract even for exempted from release
+/// objects.
+///
+/// Noop if [kFlutterMemoryAllocationsEnabled] is false.
+///
+/// The method is useful for creating singletons:
+///
+/// ```dart
+/// class MyDisposableClass {}
+///
+/// final MyDisposableClass mySingleton = exemptFromRelease(() => exemptFromRelease());
+/// ```
+T exemptFromRelease<T>(ObjectBuilderCallback<T> builder){
```

Comment:
Also, this probably doesn't belong into the global name space. Maybe it should be a static on FlutterMemoryAllocations?

Reply:
Moved to global namespace responding to this comment: https://github.com/flutter/flutter/pull/142870#discussion_r1478614887